api/server/server.go

```diff
@@ -974,11 +975,28 @@ func whoAmI() net.IP {
 func extractFields(c *gin.Context) logrus.Fields {
 	fields := logrus.Fields{"action": path.Base(c.HandlerName())}
 	for _, param := range c.Params {
-		fields[param.Key] = param.Value
+		formattedLog := camelCaseToUnderscore(param.Key)
+		fields[formattedLog] = param.Value
 	}
 	return fields
 }
 
+var camel = regexp.MustCompile("(^[^A-Z0-9]*|[A-Z0-9]*)([A-Z0-9][^A-Z]+|$)")
+
+// converts a camelCase string to lowercase_and_underscore
+func camelCaseToUnderscore(s string) string {
+	var newField []string
+	for _, sub := range camel.FindAllStringSubmatch(s, -1) {
```

Comment:
this is going to run on every request (multiple times, for certain paths) so I'd like to make sure that this function runs with minimal overhead. regex is going to be very slow to match this, go's regex isn't very cheap generally and this query is really simple to replace without using regex. i'd say that we should run through these and find the indices to replace and then create one string with the replacements via concatenation, but honestly we should just change all the url params located in this file to be snake case eg `fn_id` instead of `fnID` -- the params are only used by our code, we should just make them what we need and not do any string replacement - we probably should have made them all snake case a long time ago, anyway, so that seems like the best plan of action to me...

Reply:
@rdallman I agree, there was some confusion over this and so I am reverting back to a previous commit.